internal/apis/kfd/v1alpha2/eks/create/infrastructure.go

```diff
@@ -103,14 +108,35 @@ func (i *Infrastructure) Exec() error {
 		return fmt.Errorf("error running terraform init: %w", err)
 	}
 
-	if err := i.tfRunner.Plan(timestamp); err != nil {
+	plan, err := i.tfRunner.Plan(timestamp)
+	if err != nil {
 		return fmt.Errorf("error running terraform plan: %w", err)
 	}
 
 	if i.dryRun {
 		return nil
 	}
 
+	tfParser := parser.NewTfPlanParser(plan)
+
+	parsedPlan := tfParser.Parse()
+
+	eksInf := eks.NewInfra()
+
+	criticalResources := slices.Intersection(eksInf.GetCriticalTFResources(), parsedPlan.Destroy)
+
+	if len(criticalResources) > 0 {
```

Comment:
IIRC we have a `--force` flag that makes furyctl not ask for confirmation so we can run the delete command with automation (a script for example), maybe we should have something similar here?

A `create` operation that is destroying critical resources should not be taken lightly. So at first glance, I'd say we shouldn't add a flag to skip this check, but what do you think? Have you thought about this?

Reply:
yep, I think making --force also skip this check would be worse than lock a script, maybe in the future we can think of a "--ci" flag to instead return an error instead of prompting the user